File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/robotSpecific/targetLanguage/ev3c4ev3/control_logic.cpp
#define PROGRAM_NAME "NEPOprog"
#define WHEEL_DIAMETER 5.6
#define TRACK_WIDTH 18.0

#include <ev3.h>
#include <math.h>
#include <list>
#include "NEPODefs.h"


void ____control();
void ____logic();

double ___numberVar = 0;
bool ___booleanVar = true;
std::string ___stringVar = "";
Color ___colourVar = White;
BluetoothConnectionHandle ___connectionVar = NULL;
std::list<double> ___numberList = ((std::list<double>){0, 0});
std::list<bool> ___booleanList = ((std::list<bool>){true, true});
std::list<std::string> ___stringList = ((std::list<std::string>){"", ""});
std::list<Color> ___colourList = ((std::list<Color>){White, White});
std::list<BluetoothConnectionHandle> ___connectionList = ((std::list<BluetoothConnectionHandle>){___connectionVar, ___connectionVar});

void ____control() {
    if ( ___booleanVar ) {
    } else if ( ___booleanVar ) {
    }
    if ( ___booleanVar ) {
    } else if ( ___booleanVar ) {
    }
    while ( true ) {
    }
    for (float ___k0 = 0; ___k0 < ___numberVar; ___k0 += 1) {
    }
    for (float ___i = ___numberVar; ___i < ___numberVar; ___i += ___numberVar) {
    }
    while ( true ) {
        break;
    }
    while ( true ) {
        continue;
    }
    Wait(___numberVar);
    while ( ___booleanVar ) {
    }
    while ( ! ___booleanVar ) {
    }
    double ___item;
    for(int i = 0; i < ___numberList.size(); ++i) {
        double ___item = _getListElementByIndex(___numberList, i);
    }
    bool ___item2;
    for(int i = 0; i < ___booleanList.size(); ++i) {
        bool ___item2 = _getListElementByIndex(___booleanList, i);
    }
    std::string ___item3;
    for(int i = 0; i < ___stringList.size(); ++i) {
        std::string ___item3 = _getListElementByIndex(___stringList, i);
    }
    Color ___item4;
    for(int i = 0; i < ___colourList.size(); ++i) {
        Color ___item4 = _getListElementByIndex(___colourList, i);
    }
    BluetoothConnectionHandle ___item5;
    for(int i = 0; i < ___connectionList.size(); ++i) {
        BluetoothConnectionHandle ___item5 = _getListElementByIndex(___connectionList, i);
    }
    while ( true ) {
        if ( ___booleanVar ) {
            break;
        }
        if ( ___booleanVar ) {
            break;
        }
    }
    while ( true ) {
        if ( ___booleanVar ) {
            break;
        }
    }
}

void ____logic() {
    DrawString(ToString(___numberVar == ___numberVar), ___numberVar, ___numberVar);
    DrawString(ToString(___numberVar != ___numberVar), ___numberVar, ___numberVar);
    DrawString(ToString(___numberVar < ___numberVar), ___numberVar, ___numberVar);
    DrawString(ToString(___numberVar <= ___numberVar), ___numberVar, ___numberVar);
    DrawString(ToString(___numberVar > ___numberVar), ___numberVar, ___numberVar);
    DrawString(ToString(___numberVar >= ___numberVar), ___numberVar, ___numberVar);
    DrawString(ToString(___booleanVar && ___booleanVar), ___numberVar, ___numberVar);
    DrawString(ToString(___booleanVar || ___booleanVar), ___numberVar, ___numberVar);
    DrawString(ToString(! ___booleanVar), ___numberVar, ___numberVar);
    DrawString(ToString(true), ___numberVar, ___numberVar);
    DrawString(ToString(false), ___numberVar, ___numberVar);
    DrawString(ToString(NULL), ___numberVar, ___numberVar);
    DrawString(ToString(( ( ___booleanVar ) ? ( ___numberVar ) : ( ___numberVar) )), ___numberVar, ___numberVar);
}

int main () {
    NEPOInitEV3();
    NEPOSetAllSensors(NULL, NULL, NULL, NULL);
    startLoggingThread(OUT_BC);


    ____control();
    ____logic();

    NEPOFreeEV3();
    return 0;
}

File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/robotSpecific/targetLanguage/ev3lejosv1/control_logic.java
package generated.main;

import de.fhg.iais.roberta.runtime.*;
import de.fhg.iais.roberta.runtime.ev3.*;

import de.fhg.iais.roberta.mode.general.*;
import de.fhg.iais.roberta.mode.action.*;
import de.fhg.iais.roberta.mode.sensor.*;
import de.fhg.iais.roberta.mode.action.ev3.*;
import de.fhg.iais.roberta.mode.sensor.ev3.*;

import de.fhg.iais.roberta.components.*;

import java.util.LinkedHashSet;
import java.util.HashMap;
import java.util.Set;
import java.util.Map;
import java.util.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;

import lejos.remote.nxt.NXTConnection;

public class NEPOprog {
    private static Configuration brickConfiguration;

    private Set<UsedSensor> usedSensors = new LinkedHashSet<UsedSensor>();
    private Hal hal = new Hal(brickConfiguration, usedSensors);

    private void ____control() {
        if ( ___booleanVar ) {
        } else if ( ___booleanVar ) {
        }
        if ( ___booleanVar ) {
        } else if ( ___booleanVar ) {
        }
        if ( true ) {
            while ( true ) {
            }
        }
        for ( float ___k0 = 0; ___k0< ___numberVar; ___k0+= 1 ) {
        }
        for ( float ___i = ___numberVar; ___i< ___numberVar; ___i+= ___numberVar ) {
        }
        if ( true ) {
            while ( true ) {
                break;
            }
        }
        if ( true ) {
            while ( true ) {
                continue;
            }
        }
        hal.waitFor(___numberVar);
        while ( ___booleanVar ) {
        }
        while ( ! ___booleanVar ) {
        }
        for ( float ___item : ___numberList ) {
        }
        for ( boolean ___item2 : ___booleanList ) {
        }
        for ( String ___item3 : ___stringList ) {
        }
        for ( PickColor ___item4 : ___colourList ) {
        }
        for ( NXTConnection ___item5 : ___connectionList ) {
        }
        while ( true ) {
            if ( ___booleanVar ) {
                break;
            }
            if ( ___booleanVar ) {
                break;
            }
            hal.waitFor(15);
        }
        while ( true ) {
            if ( ___booleanVar ) {
                break;
            }
            hal.waitFor(15);
        }
    }

    private void ____logic() {
        hal.drawText(String.valueOf(___numberVar == ___numberVar), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(___numberVar != ___numberVar), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(___numberVar < ___numberVar), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(___numberVar <= ___numberVar), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(___numberVar > ___numberVar), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(___numberVar >= ___numberVar), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(___booleanVar && ___booleanVar), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(___booleanVar || ___booleanVar), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(! ___booleanVar), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(true), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(false), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(null), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(( ( ___booleanVar ) ? ___numberVar : ___numberVar )), ___numberVar, ___numberVar);
    }

    public static void main(String[] args) {
        try {
             brickConfiguration = new EV3Configuration.Builder()
                .setWheelDiameter(5.6)
                .setTrackWidth(18.0)
                .build();

            new NEPOprog().run();
        } catch ( Exception e ) {
            Hal.displayExceptionWaitForKeyPress(e);
        }
    }

    float ___numberVar = 0;
    boolean ___booleanVar = true;
    String ___stringVar = "";
    PickColor ___colourVar = PickColor.WHITE;
    NXTConnection ___connectionVar = null;
    ArrayList<Float> ___numberList = new ArrayList<>(Arrays.asList((float) 0, (float) 0));
    ArrayList<Boolean> ___booleanList = new ArrayList<>(Arrays.<Boolean>asList(true, true));
    ArrayList<String> ___stringList = new ArrayList<>(Arrays.<String>asList("", ""));
    ArrayList<PickColor> ___colourList = new ArrayList<>(Arrays.<PickColor>asList(PickColor.WHITE, PickColor.WHITE));
    ArrayList<NXTConnection> ___connectionList = new ArrayList<>(Arrays.<NXTConnection>asList(___connectionVar, ___connectionVar));

    public void run() throws Exception {
        hal.startLogging();
        ____control();
        ____logic();
        hal.closeResources();

    }
}